lyngk.js

```diff
@@ -0,0 +1,179 @@
+    'R': "Red",
+    'K': "Black",
+    'B': "Blue",
+    'I': "Ivory",
+    'G': "Green",
+    'J': "White",
+    '-': "Empty"
+  };
+
+  my.startUp = drawBoard();
+
+  function drawBoard() {
+    ctx.clearRect(0, 0, c.width, c.height);
+    var colorCount = {
+      Ivory: 8,
+      Blue: 8,
+      Red: 8,
+      Green: 8,
+      Black: 8,
+      White: 3
+    };
+    //var shuffled = shuffleArray(nodeList);
+
+    for (var i = 0; i < verticalX.length; i++) {
+      drawLine((xRoot * verticalX[i] + padding), (yOffset * verticalY1[i] + padding), (xRoot * verticalX[i] + padding), (yOffset * verticalY2[i] + padding));
+    }
+    for (var j = 0; j < diagonalDestination.length; j++) {
+      drawLine((xRoot * diagonalSourceCoords[j][0] + padding), (yOffset * diagonalSourceCoords[j][1] + padding), (xRoot * diagonalDestination[j][0] + padding), (yOffset * diagonalDestination[j][1] + padding));
+    }
+    /*for (var k = 0; k < nodeList.length; k++) {
+      var coordinates = boardPosition[nodeList[k]];
+      if (k < 8) {
+        drawGamePiece((xRoot * coordinates[0] + padding), (yOffset * coordinates[1] + padding), "Ivory");
+        colorCount.Ivory = colorCount.Ivory - 1;
+      }
+      else if (k < 16) {
+        drawGamePiece((xRoot * coordinates[0] + padding), (yOffset * coordinates[1] + padding), "Blue");
+        colorCount.Blue = colorCount.Blue - 1;
+      }
+      else if (k < 24) {
+        drawGamePiece((xRoot * coordinates[0] + padding), (yOffset * coordinates[1] + padding), "Red");
+        colorCount.Red = colorCount.Red - 1;
+      }
+      else if (k < 32) {
+        drawGamePiece((xRoot * coordinates[0] + padding), (yOffset * coordinates[1] + padding), "Green");
+        colorCount.Green = colorCount.Green - 1;
+      }
+      else if (k < 40) {
+        drawGamePiece((xRoot * coordinates[0] + padding), (yOffset * coordinates[1] + padding), "Black");
+        colorCount.Black = colorCount.Black - 1;
+      }
+      else {
+        drawGamePiece((xRoot * coordinates[0] + padding), (yOffset * coordinates[1] + padding), "White");
+        colorCount.White = colorCount.White - 1;
+      }
+    }
+    stackGamePiece((xRoot * boardPosition['E4'][0] + padding), (yOffset * boardPosition['E4'][1] + padding), 'Black', 2);
+    stackGamePiece((xRoot * boardPosition['E4'][0] + padding), (yOffset * boardPosition['E4'][1] + padding), 'Blue', 3);
+    stackGamePiece((xRoot * boardPosition['E4'][0] + padding), (yOffset * boardPosition['E4'][1] + padding), 'Green', 4);
+    stackGamePiece((xRoot * boardPosition['E4'][0] + padding), (yOffset * boardPosition['E4'][1] + padding), 'White', 5);
+    stackGamePiece((xRoot * boardPosition['E4'][0] + padding), (yOffset * boardPosition['E4'][1] + padding), 'Ivory', 6);*/
+  }
+
+  function drawLine(x1, y1, x2, y2) {
+	  ctx.beginPath();
+	  ctx.moveTo(x1, y1);
+	  ctx.lineTo(x2, y2);
+	  ctx.stroke();
+  }
+
+  function drawGamePiece(xcenter, ycenter, color) {
+    ctx.beginPath();

```

Comment:
Leaving ctx in a member variable is better than retrieving it every time, but it would be even better to hand it in as an explicit argument because you could imagine a case in which there are two boards onone screen which would have different 2d contexts.